### PR TITLE
HLRC: ML Add preview datafeed api

### PR DESCRIPTION
**Pull request by benwtrent**
This adds the ML preview datafeed API to the HLRC

Relates to #29827

### CODE REVIEW

File: docs/java-rest/high-level/ml/preview-datafeed.asciidoc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+==== Execution
+
+The request can be executed through the `MachineLearningClient` contained
+in the `RestHighLevelClient` object, accessed via the `machineLearningClient()` method.
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/MlClientDocumentationIT.java[x-pack-ml-preview-datafeed-execute]
+--------------------------------------------------
+<1> The raw BytesReference of the data preview
```
Comment (droberts195) [inline docs/java-rest/high-level/ml/preview-datafeed.asciidoc]: BytesReference would look better in backticks: `BytesReference`

File: docs/java-rest/high-level/ml/preview-datafeed.asciidoc [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+<2> A List of Map<String,Object> that represents the previewed data
```
Comment (droberts195) [inline docs/java-rest/high-level/ml/preview-datafeed.asciidoc]: Similarly `List` and `Map<String, Object>` would look better in backticks.

Also, I think we usually put a space after the comma in `Map<String, Object>`.  It's not so important for private code but it's nice to make the public docs completely consistent.

File: docs/java-rest/high-level/ml/preview-datafeed.asciidoc [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+The method does not block and returns immediately. The passed `ActionListener` is used
+to notify the caller of completion. A typical `ActionListener` for `PreviewDatafeedResponse` may
+look like
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/MlClientDocumentationIT.java[x-pack-ml-preview-datafeed-listener]
+--------------------------------------------------
+<1> `onResponse` is called back when the action is completed successfully
+<2> `onFailure` is called back when some unexpected error occurs
```
Comment (droberts195) [inline docs/java-rest/high-level/ml/preview-datafeed.asciidoc]: Is it possible that we can reuse the stuff that was added in #34125 to cut out the duplication in this async section?

**Review by droberts195 — COMMENTED** [review]
I left a couple of nits on the docs and also something worth investigating for whether we can reduce the duplication in the docs.  If it is possible to reduce the duplication and you'd rather leave that for a followup PR that applies to change to all our docs then that's fine.

File: docs/java-rest/high-level/ml/preview-datafeed.asciidoc (hunk 3 above)
Comment (benwtrent) [reply docs/java-rest/high-level/ml/preview-datafeed.asciidoc]: I did not know that was a thing! will update this PR and my other HLRC one pronto

**Review by droberts195 — APPROVED** [review]
LGTM

**Pull request merged**